Yank/storage/iodrivers.py

```diff
@@ -0,0 +1,1753 @@
+                    # Check if storage object IS a group (e.g. dict)
+                    try:
+                        obj = head_group.groups[split_path[-1]]
+                        store_type = obj.getncattr('IODriver_Storage_Type')
+                        if store_type == 'groups':
+                            variable = obj
+                            is_group = True
+                    except AttributeError:  # Trap the case of no group name in head_group, non-fatal
+                        pass
+                if not is_group:
+                    # Bind to the specific variable instead since its not a group
+                    variable = head_group.variables[split_path[-1]]
+            except KeyError:
+                raise KeyError("No variable found at {} on file!".format(path))
+            try:
+                # Bind to the storage type by mapping IODriver_Type -> Known TypeHandler
+                data_type = variable.getncattr('IODriver_Type')
+                head_path = '/'.join(split_path[:-1])
+                target_name = split_path[-1]
+                # Remember the group for the future while also getting storage binder
+                if head_path == '':
+                    storage_object = self.ncfile
+                else:
+                    storage_object = self._bind_group(head_path)
+                uninstanced_handler = self._IOMetaDataReaders[data_type]
+                self._variables[path] = uninstanced_handler(self, target_name, storage_object=storage_object)
+                handler = self._variables[path]
+            except AttributeError:
+                raise AttributeError("Cannot auto-detect variable type, ensure that 'IODriver_Type' is a set ncattr")
+            except KeyError:
+                raise KeyError("No mapped type handler known for 'IODriver_Type' = '{}'".format(data_type))
+        return handler
+
+    def create_storage_variable(self, path, type_key):
+        path = normalize_path(path)
+        try:
+            handler = self._type_maps[type_key]
+        except KeyError:
+            raise KeyError("No known Type Handler for given type!")
+        split_path = decompose_path(path)
+        # Bind groups as needed, splitting off the last entry
+        head_path = '/'.join(split_path[:-1])
+        target_name = split_path[-1]
+        if head_path == '':
+            storage_object = self.ncfile
+        else:
+            storage_object = self._bind_group(head_path)
+        self._variables[path] = handler(self, target_name, storage_object=storage_object)
+        return self._variables[path]
+
+    @staticmethod
+    def check_scalar_dimension(ncfile):
+        """
+        Check that the `scalar` dimension exists on file and create it if not
+
+        """
+        if 'scalar' not in ncfile.dimensions:
+            ncfile.createDimension('scalar', 1)  # scalar dimension
+
+    @staticmethod
+    def check_infinite_dimension(ncfile, name='iteration'):
```

Comment:
I want to leave this in since people may actually use it as part of the standalone aspect.